raianalysis/raianalysis/raianalyzer/raianalyzer.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the RAIAnalyzer class."""
+
+class RAIAnalyzer(object):
+    def __init__(self, model, train, test, target_column, transformation, task_type):
+        self.model = model
+        self.train = train
+        self.test = test
+        self.target_column = target_column
+        self.transformation = transformation
+        self.task_type = task_type
+
+    def add_fairness(self, sensitive_features):
+        pass
+
+    def add_explainer(self):
```

Comment:
I am concerned that these are going to end up having monstrously long argument lists as we gradually add features.

Reply:
+1

Reply:
That was the motivation behind the 'managers' - each component could present its own API as it saw fit. For explanations, this might mean having separate `add_mimic_explainer()` and `add_lime_explainer()` methods, keeping the individual argument lists to within reasonable bounds. The reason for the extra layer of managers was to group related functionality together, and keep the API surface of the `RAIAnalyzer` object reasonably small.

The alternative proposed yesterday of having 'Assets' (not AML Assets... I don't like overloading the name) would also work in a similar way. We could start with a basic `ExplainerAsset` which had few, if any, arguments, but then have a `LIMEExplainerAsset` or `MimicExplainerAsset` etc. as required.

Reply:
for explanations the API should be really simple for now.  In future I can see either passing in an explainer object or doing something like 
```
add_explainer(MimicExplainerThing(...))
```

Reply:
I agree it would be too much to have methods like add_mimic_explainer() and add_lime_explainer() for each individual thing

Reply:
I'll go with `Thing` rather than `Asset` :-) I feel that that would keep things extensible and coherent.

Reply:
punting this to a later time, resolving for now